fibers.cpp

```diff
@@ -238,20 +247,20 @@ bool Fibers::yield()
 		return true;
 		}
 
-	for (int i = 1; i < MAXFIBERS; ++i)
+	fi = fi % (MAXFIBERS - 1) + 1;
+	if (runnable(fibers[fi]))

```

Comment:
Why did this get moved outside the loop? A loop is necessary to search for a runnable fiber. 

Only cleaning up the ones we scan over is a way of limiting the cleanup work to keep the yield overhead low. It's not critical to clean up everything all at once. Either yield or create will eventually handle it.